README.md

```diff
@@ -44,3 +47,32 @@ You can either use a premade binary from the [releases page](https://github.com/
 ```sh
 go get -u github.com/Code0x58/gitlab-ci-validate
 ```
+
+#### Usage with Docker containers
+You can use the Dockerfile to build your own image or use the pre-built version available at the [Gitlab container registry](https://gitlab.com/comedian780/docker-gitlab-ci-validate/container_registry).
+
+You can run tests against the gitlab.com endpoint:  
+If no parameter is given the container will look for a file called `.gitlab-ci.yml`
+```sh
+docker run -i --rm \
+-v ${PWD}/.gitlab-ci.yml:/yaml/.gitlab-ci.yml \
+registry.gitlab.com/comedian780/docker-gitlab-ci-validate
+```
+
+You can run tests against a self hosted Gitlab instance with custom filenames:  
+Set the credentials and URL via the `GITLAB_HOST` environment variable  
+```sh
+docker run -i --rm \
+-e GITLAB_HOST=https://GITLAB_USER:GITLAB_PW@your.gitlab.server \
+-v ${PWD}:/yaml \
+-v /additional/folder/.additional.yml:/yaml/.additional.yml \
+registry.gitlab.com/comedian780/docker-gitlab-ci-validate custom.yml .files.yaml .additional.yml
+```
+
+You can also test all YAML files inside a directory (this also includes YAML files in subdirectories):
```

Comment:
I feel this is a little over-reaching as `gitlab-ci-validate` will reject anything not ending with `.yml`, and I imagine scenarios where this would be relevant would be rare

_p.s. just because code golf:_
```bash
find . -type f -iname \*.yml | xargs docker run -i --rm -v $PWD:/yaml registry.gitlab.com/comedian780/docker-gitlab-ci-validate
```